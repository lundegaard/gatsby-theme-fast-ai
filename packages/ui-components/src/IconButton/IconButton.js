import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';

import Box from '../Box';

const IconButton = forwardRef(({ sx, size = 32, ...rest }, ref) => (
	<Box
		as="button"
		ref={ref}
		{...rest}
		sx={{
			alignItems: 'center',
			appearance: 'none',
			bg: 'transparent',
			border: 'none',
			borderRadius: 4,
			color: 'inherit',
			display: 'inline-flex',
			height: size,
			justifyContent: 'center',
			padding: 1,
			width: size,
			...sx,
		}}
	/>
));

IconButton.displayName = 'IconButton';
IconButton.propTypes = {
	size: PropTypes.number,
	sx: PropTypes.object,
};

export default IconButton;
