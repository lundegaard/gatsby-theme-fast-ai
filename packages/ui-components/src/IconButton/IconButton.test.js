import React from 'react';
import { shallow } from 'enzyme';

import IconButton from './IconButton';

describe('<IconButton />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<IconButton {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
