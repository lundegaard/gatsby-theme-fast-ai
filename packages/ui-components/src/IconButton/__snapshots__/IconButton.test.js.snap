// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> matches snapshot when shallowly rendered 1`] = `
<Box
  as="button"
  sx={
    {
      "alignItems": "center",
      "appearance": "none",
      "bg": "transparent",
      "border": "none",
      "borderRadius": 4,
      "color": "inherit",
      "display": "inline-flex",
      "height": 32,
      "justifyContent": "center",
      "padding": 1,
      "width": 32,
    }
  }
/>
`;
